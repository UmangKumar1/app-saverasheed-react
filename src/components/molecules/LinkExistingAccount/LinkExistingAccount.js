import React from 'react'
import PropTypes from 'prop-types'
import styled, { css } from 'react-emotion'

import muiTheme from '../../../theme'
import Button from '../../atoms/Button/Button'

const { string, func } = PropTypes

const {
  palette: {
    tertiary: { main: tertiaryMain, contrastText },
  },
} = muiTheme

const AuthButtonsRoot = styled('div')({
  display: 'flex',
  flexDirection: 'column',
  color: contrastText,
  marginBottom: 50,
})

const AuthButtonsTitle = styled('h1')({
  textAlign: 'center',
  marginBottom: 30,
})

const buttonCx = css({
  color: tertiaryMain,
  padding: 8,
  justifyContent: 'center',
  fontSize: 24,
  marginTop: 16,
})

const LinkExistingAccount = ({ method, email, pendingCredId, linkAccount }) => (
  <AuthButtonsRoot>
    <AuthButtonsTitle>Account Exists</AuthButtonsTitle>
    <div>
      An account with the email {email} exists with {method}. Do you want to
      link this account with {pendingCredId}?
    </div>
    <Button className={buttonCx} onClick={linkAccount}>
      Link Account
    </Button>
  </AuthButtonsRoot>
)

LinkExistingAccount.propTypes = {
  method: string.isRequired,
  email: string.isRequired,
  pendingCredId: string.isRequired,
  linkAccount: func.isRequired,
}

export default LinkExistingAccount
