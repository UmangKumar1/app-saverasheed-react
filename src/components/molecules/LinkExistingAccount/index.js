export { default, default as LinkExistingAccount } from './LinkExistingAccount'
