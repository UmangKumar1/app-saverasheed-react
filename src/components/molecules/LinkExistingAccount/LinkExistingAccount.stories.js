import React from 'react'
import { storiesOf } from '@storybook/react'
import { action } from '@storybook/addon-actions'

import LinkExistingAccount from '.'

storiesOf('LinkExistingAccount', module).add('sign in', () => (
  <LinkExistingAccount
    method="facebook.com"
    pendingCredId="google.com"
    email="rasheed.bustamam@gmail.com"
    linkAccount={action('link!')}
  />
))
