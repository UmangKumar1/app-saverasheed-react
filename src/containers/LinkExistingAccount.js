import { connect } from 'react-redux'

import LinkExistingAccount from '../components/molecules/LinkExistingAccount'
import { linkExistingAccountRequest } from '../modules/auth/actions'
import {
  getEmail,
  getMethod,
  getPendingCredId,
} from '../modules/auth/selectors'
import { subscribe } from '../util/reselect'

const mapStateToProps = subscribe({
  getMethod,
  getPendingCredId,
  getEmail,
})

const actions = { linkAccount: linkExistingAccountRequest }

export default connect(
  mapStateToProps,
  actions,
)(LinkExistingAccount)
