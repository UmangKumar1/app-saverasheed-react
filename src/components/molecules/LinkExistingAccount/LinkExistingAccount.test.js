import React from 'react'
import { shallow } from 'enzyme'

import LinkExistingAccount from '.'

const wrap = (props = {}) => shallow(<LinkExistingAccount {...props} />)

describe('LinkExistingAccount', () => {
  it('renders children when passed in', () => {
    const wrapper = wrap({
      method: 'facebook.com',
      pendingCredId: 'google.com',
      email: 'rasheed.bustamam@gmail.com',
      linkAccount: jest.fn(),
    })
    expect(wrapper).toMatchSnapshot()
  })
})
