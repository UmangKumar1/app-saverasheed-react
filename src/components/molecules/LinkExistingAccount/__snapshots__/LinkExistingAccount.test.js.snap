// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LinkExistingAccount renders children when passed in 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <LinkExistingAccount
    email="rasheed.bustamam@gmail.com"
    linkAccount={[MockFunction]}
    method="facebook.com"
    pendingCredId="google.com"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": Array [
        <Styled(h1)>
          Account Exists
        </Styled(h1)>,
        <div>
          An account with the email 
          rasheed.bustamam@gmail.com
           exists with 
          facebook.com
          . Do you want to link this account with 
          google.com
          ?
        </div>,
        <Styled(button)
          className="css-kz4gx8"
          onClick={[MockFunction]}
        >
          Link Account
        </Styled(button)>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": "Account Exists",
        },
        "ref": null,
        "rendered": "Account Exists",
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "An account with the email ",
            "rasheed.bustamam@gmail.com",
            " exists with ",
            "facebook.com",
            ". Do you want to link this account with ",
            "google.com",
            "?",
          ],
        },
        "ref": null,
        "rendered": Array [
          "An account with the email ",
          "rasheed.bustamam@gmail.com",
          " exists with ",
          "facebook.com",
          ". Do you want to link this account with ",
          "google.com",
          "?",
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": "Link Account",
          "className": "css-kz4gx8",
          "onClick": [MockFunction],
        },
        "ref": null,
        "rendered": "Link Account",
        "type": [Function],
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <Styled(h1)>
            Account Exists
          </Styled(h1)>,
          <div>
            An account with the email 
            rasheed.bustamam@gmail.com
             exists with 
            facebook.com
            . Do you want to link this account with 
            google.com
            ?
          </div>,
          <Styled(button)
            className="css-kz4gx8"
            onClick={[MockFunction]}
          >
            Link Account
          </Styled(button)>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": "Account Exists",
          },
          "ref": null,
          "rendered": "Account Exists",
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "An account with the email ",
              "rasheed.bustamam@gmail.com",
              " exists with ",
              "facebook.com",
              ". Do you want to link this account with ",
              "google.com",
              "?",
            ],
          },
          "ref": null,
          "rendered": Array [
            "An account with the email ",
            "rasheed.bustamam@gmail.com",
            " exists with ",
            "facebook.com",
            ". Do you want to link this account with ",
            "google.com",
            "?",
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": "Link Account",
            "className": "css-kz4gx8",
            "onClick": [MockFunction],
          },
          "ref": null,
          "rendered": "Link Account",
          "type": [Function],
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
